cmd/operator-sdk/test/local.go

```diff
@@ -293,6 +243,162 @@ func testLocalGoFunc(cmd *cobra.Command, args []string) error {
 	return nil
 }
 
+func collectManifests() error {
+	// Pass empty files to the test-framework if no setup required
+	if tlConfig.noSetup {
+		emptyBytes := []byte{}
+		file, err := writeToTempFile(emptyBytes, "empty.yaml")
+		if err != nil {
+			return err
+		}
+		tlConfig.namespacedManPath = file.Name()
+		tlConfig.globalManPath = file.Name()
+
+		f := func() {
+			err := os.Remove(file.Name())
+			if err != nil {
+				log.Errorf("Could not delete empty manifest file: (%v)", err)
+			}
+		}
+		cleanups = append(cleanups, f)
+		return nil
+	}
+
+	// Collect manifest from deploy dir if legacy layout
+	// TODO: Remove this after the legacy layout has been deprecated
+	if !kbutil.HasProjectFile() {
+		return legacyCollectFromDeployDir()
+	}
+
+	// For new layout check if the manifests are piped to stdin
+	stdinF, err := os.Stdin.Stat()
+	if err != nil {
+		return err
+	}
+	if stdinF.Mode()&os.ModeNamedPipe != 0 {
+		return collectFromStdin()
+	}
+
+	// If no input is piped to stdin, the manifest paths must be set by flags
+	if tlConfig.namespacedManPath == "" || tlConfig.globalManPath == "" {
+		return fmt.Errorf("The namespaced and global manifests must be piped to stdin" +
+			" e.g (kustomize build config/default | operator-sdk test ...).\n" +
+			"Or the both the flags --namespaced-manifest and --global-manifest must be set.")
+	}
+	return nil
+}
+
+func collectFromStdin() error {
+	reader := bufio.NewReader(os.Stdin)
+	b, err := ioutil.ReadAll(reader)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	namespacedManifests := []byte{}
+	globalManifests := []byte{}
+	scanner := internalk8sutil.NewYAMLScanner(b)
+	for scanner.Scan() {
+		manifest := scanner.Bytes()
+		typeMeta, err := internalk8sutil.GetTypeMetaFromBytes(manifest)
+		if err != nil {
+			log.Debugf("No TypeMeta found, skipping file")
+			continue
+		}
+		switch typeMeta.GroupVersionKind().Kind {
+		// Namespaced scoped resources
+		case "Role", "RoleBinding", "ServiceAccount", "Deployment", "Service":
+			namespacedManifests = internalk8sutil.CombineManifests(namespacedManifests, manifest)
+
+		// Cluster scoped resources
+		case "ClusterRole", "ClusterRoleBinding", "CustomResourceDefinition":
+			globalManifests = internalk8sutil.CombineManifests(globalManifests, manifest)
+		default:
+			// TODO: Should the framework create other manifests by checking
+			// the `namespace` field to see if it's global vs namespaced?
+			// "kustomize build" can output extraneous manifests not relevant(?) to testing,
+			// e.g ServiceMonitor, Namespace
```

Comment:
For kubebuilder layout the command 

```
# Install CRDs into a cluster
install: manifests kustomize
	$(KUSTOMIZE) build config/crd | kubectl apply -f -
```

Will apply all. Could we not apply all instead of check is or not cluster scoped? Why it would be required?